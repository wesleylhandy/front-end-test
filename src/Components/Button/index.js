import React from "react";
import PropTypes from "prop-types";
import "./button.css";

const Button = ({ style = {}, clickHandler, text }) => (
	<button className="Button" style={style} onClick={clickHandler}>
		{text}
	</button>
);

Button.propTypes = {
	style: PropTypes.object,
	clickHandler: PropTypes.func,
	text: PropTypes.string.isRequired,
};

export default Button;
