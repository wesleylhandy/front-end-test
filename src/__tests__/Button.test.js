import React from "react";
import Button from "../Components/Button";
import { create, act } from "react-test-renderer";

it("renders correctly", () => {
	const tree = create(
		<Button
			text={"Initial Value"}
			clickHandler={() => jest.fn()}
			style={{ color: "white", background: "black" }}
		/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});

it("displays the right text when the props is updated", () => {
	const tree = create(<Button text={"Some Other Text"} />).toJSON();
	expect(tree).toMatchSnapshot();
});

it("updates component style when passed a style object", () => {
	const tree = create(
		<Button style={{ color: "red", background: "white" }} text="" />
	).toJSON();
	expect(tree).toMatchSnapshot();
});

it("calls the clickHandler function when user clicks on button", () => {
	let component;
	act(() => {
		component = create(<Button clickHandler={() => "clicked"} text="" />);
	});

	const instance = component.root;
	const button = instance.findByType("button");
	let result;
	act(() => {
		result = button.props.onClick();
	});
	expect(result).toBe("clicked");
});
