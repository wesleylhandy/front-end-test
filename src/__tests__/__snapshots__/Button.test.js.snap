// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays the right text when the props is updated 1`] = `
<button
  className="Button"
  onClick={undefined}
  style={Object {}}
>
  Some Other Text
</button>
`;

exports[`renders correctly 1`] = `
<button
  className="Button"
  onClick={[Function]}
  style={
    Object {
      "background": "black",
      "color": "white",
    }
  }
>
  Initial Value
</button>
`;

exports[`updates component style when passed a style object 1`] = `
<button
  className="Button"
  onClick={undefined}
  style={
    Object {
      "background": "white",
      "color": "red",
    }
  }
>
  
</button>
`;
